pkg/provider/azure.go

```diff
@@ -640,7 +640,7 @@ func (az *Cloud) configAzureClients(
 	az.FileClient = fileclient.New(fileClientConfig)
 }
 
-func (az *Cloud) getAzureClientConfig(servicePrincipalToken *adal.ServicePrincipalToken) *azclients.ClientConfig {
+func (az *Cloud) getAzureClientConfig(servicePrincipalToken adal.OAuthTokenProvider) *azclients.ClientConfig {
```

Comment:
I'm not sure this is going to work, seems to be causing tests to fail... need to look into it a bit more